### PR TITLE
Edit FRIDA mode and QEMU mode READMEs

### PR DESCRIPTION
**Pull request by llzmb**
Changes:
- Fix typos.
- Fix spelling.
- Fix references.
- Fix formatting.
- Fix line length.

### CODE REVIEW

File: frida_mode/README.md [hunk 1]
```diff
@@ -1,38 +1,40 @@
...
-AFL just like that provided by QEMU mode. The intention is to provide a very
+AFL++ just like that provided by QEMU mode. The intention is to provide a very
 similar user experience, right down to the options provided through environment
 variables.
 
-Whilst AFLplusplus already has some support for running on FRIDA [here](https://github.com/AFLplusplus/AFLplusplus/tree/stable/utils/afl_frida)
+Whilst AFL++ already has some support for running on FRIDA
+[here](https://github.com/AFLplusplus/AFLplusplus/tree/stable/utils/afl_frida),
 this requires the code to be fuzzed to be provided as a shared library, it
 cannot be used to fuzz executables. Additionally, it requires the user to write
```
Comment (llzmb) [inline frida_mode/README.md]: The link is broken and the paragraph seems to be outdated. Can I delete the paragraph?

Is FRIDA mode still a WIP or is it as far developed as QEMU mode?

File: frida_mode/README.md [hunk 2]
```diff
@@ -168,200 +172,209 @@ Generated block 0x7ffff75e98e2
...
+function marked `__attribute__((constructor()))` or the test harness can simply
+be linked against it. It is important that the target library is loaded before
+execution of `main`, since this is the point where FRIDA mode is initialized.
+Otherwise, it will not be possible to configure coverage for the test library
+using `AFL_FRIDA_INST_RANGES` or similar.
+
+## Debugging
+
+Please refer to the [debugging](#debugging) guide for assistance should you
 encounter problems with FRIDA mode.
```
Comment (llzmb) [inline frida_mode/README.md]: The link doesn't point anywhere. Should there be a debugging guide in this README or outside of this README?

**Comment by llzmb** [comment]
How to write _FRIDA mode_?

I found:
- FRIDA MODE
- FRIDA mode
- Frida mode
- frida_mode
- frida-mode

Do you want to distinguish between plain text (maybe "FRIDA mode") and technical context (maybe "frida_mode")?

Same question for _QEMU mode_.

File: frida_mode/README.md [hunk 3]
```diff
@@ -1,38 +1,40 @@
...
+| Persistent Mode          |     x      | (x86/x64/aarch64 only)                       |
+| LAF-Intel / CompCov      |     -      | (CMPLOG is better 90% of the time)           |
+| CMPLOG                   |     x      | (x86/x64/aarch64 only)                       |
+| Selective Instrumentation|     x      |                                              |
+| Non-Colliding Coverage   |     -      | (not possible in binary-only instrumentation) |
+| Ngram prev_loc Coverage  |     -      |                                              |
+| Context Coverage         |     -      |                                              |
+| Auto Dictionary          |     -      |                                              |
+| Snapshot LKM Support     |     -      |                                              |
+| In-Memory Test Cases     |     x      | (x86/x64/aarch64 only)                       |
```
Comment (llzmb) [inline frida_mode/README.md]: Does this table have a particular order? Can I sort alphabetically (here and in other places where the feature table is used)?

Comment (vanhauser-thc) [reply frida_mode/README.md]: it is ok as is, same for the other tables, no need :)

File: frida_mode/README.md (hunk 1 above)
Comment (vanhauser-thc) [reply frida_mode/README.md]: that was a different frida implementation that I removed. you can just delete the paragraph.

File: frida_mode/README.md (hunk 2 above)
Comment (vanhauser-thc) [reply frida_mode/README.md]: @WorksButNotTested -> your feedback needed

**Comment by vanhauser-thc** [comment]
> How to write _FRIDA mode_?
> 
> I found:
> 
>     * FRIDA MODE
> 
>     * FRIDA mode
> 
>     * Frida mode
> 
>     * frida_mode
> 
>     * frida-mode
> 
> 
> Do you want to distinguish between plain text (maybe "FRIDA mode") and technical context (maybe "frida_mode")?
> 
> Same question for _QEMU mode_.

that is a good question. the directory is "frida_mode", the name is "FRIDA mode". same for QEMU, "qemu_mode" and "QEMU mode"

File: frida_mode/README.md (hunk 1 above)
Comment (llzmb) [reply frida_mode/README.md]: There are a few other mentions: https://github.com/AFLplusplus/AFLplusplus/search?q=afl_frida

I can take care of the other MD files.

Comment (llzmb) [reply frida_mode/README.md]: Resolved here. As for the other MD files, I added this to my quality assurance to do and will use a different branch.

File: qemu_mode/README.md [hunk 4]
```diff
@@ -1,172 +1,180 @@
...
 
-Just set AFL_QEMU_EXCLUDE_RANGES=A,B,C...
+Just set `AFL_QEMU_EXCLUDE_RANGES=A,B,C...`.
 
-The format of the items on the list is the same as for AFL_QEMU_INST_RANGES, and excluding ranges
-takes priority over any included ranges or AFL_INST_LIBS.
+The format of the items on the list is the same as for `AFL_QEMU_INST_RANGES`
+and excluding ranges takes priority over any included ranges or `AFL_INST_LIBS`.
 
 ## 7) CompareCoverage
```
Comment (llzmb) [inline qemu_mode/README.md:113]: Some of the info on `AFL_COMPCOV_LEVEL` can already be found in:
https://github.com/AFLplusplus/AFLplusplus/blob/dev/docs/env_variables.md#5-settings-for-afl-qemu-trace

Can I move the details from this README.md to env_variables.md?

File: frida_mode/README.md (hunk 2 above)
Comment (WorksButNotTested) [reply frida_mode/README.md]: This should point here…
https://github.com/AFLplusplus/AFLplusplus/blob/dev/frida_mode/DEBUGGING.md

**Comment by WorksButNotTested** [comment]
Many thanks. Feel free to @ me if there’s anything else you need me to look at!

File: frida_mode/README.md (hunk 2 above)
Comment (llzmb) [reply frida_mode/README.md]: Thank you, @WorksButNotTested!

File: qemu_mode/README.md (hunk 4 above)
Comment (llzmb) [reply qemu_mode/README.md:113]: Meeting minutes: Not necessary, keep as is.

File: qemu_mode/README.md [hunk 5]
```diff
@@ -1,172 +1,180 @@
...
-  (See ../README.md for the general instruction manual.)
+For the general instruction manual, see [README.md](../README.md).
 
 ## 1) Introduction
 
 The code in this directory allows you to build a standalone feature that
 leverages the QEMU "user emulation" mode and allows callers to obtain
-instrumentation output for black-box, closed-source binaries. This mechanism
-can be then used by afl-fuzz to stress-test targets that couldn't be built
-with afl-gcc.
```
Comment (vanhauser-thc) [inline qemu_mode/README.md:11]: please fix this to "afl-cc" instead of "afl-gcc" ... the readme is very outdated :) (I mean in the changed line below)

**Comment by vanhauser-thc** [comment]
just the update on our old doc and then it is ready to merge, thank you!

File: qemu_mode/README.md (hunk 5 above)
Comment (llzmb) [reply qemu_mode/README.md:11]: Is afl-gcc correct here?

> If you want to compare the performance of the QEMU instrumentation with that of
> afl-gcc compiled code against the same target, you need to build the
> non-instrumented binary with the same optimization flags that are normally
> injected by afl-gcc, and make sure that the bits to be tested are statically
> linked into the binary.

Comment (vanhauser-thc) [reply qemu_mode/README.md:11]: afl-clang-fast there too, thank you :)

**Comment by vanhauser-thc** [comment]
plus fixing the conflicts :)

**Comment by vanhauser-thc** [comment]
thank you!

**Pull request merged**